plugins/tableselection/plugin.js

```diff
@@ -536,32 +559,49 @@
 			return;
 		}
 
-		evt.stop();
-
 		selectedTable = selectedCells[ 0 ].getAscendant( 'table' );
 		selectedCells = getSelectedCells( selection, selectedTable );
 		firstCell = selectedCells[ 0 ];
 		firstRow = firstCell.getParent();
 		lastCell = selectedCells[ selectedCells.length - 1 ];
 		lastRow = lastCell.getParent();
 
-		// Empty all selected cells.
-		if ( !boundarySelection ) {
-			for ( i = 0; i < selectedCells.length; i++ ) {
-				selectedCells[ i ].setHtml( '' );
-			}
-		}
+		// Schedule selecting appropriate table cells after pasting. It covers both table and not-table
+		// content (#520).
+		editor.once( 'afterPaste', function() {
+			var toSelect = cellToPaste ?
+				getCellsBetween( new CKEDITOR.dom.element( pastedTableMap[ 0 ][ 0 ] ), cellToPaste ) :
+				selectedCells;
+
+			fakeSelectCells( editor, toSelect );
+		} );
 
 		// Handle mixed content (if the table is not the only child in the tmpContainer, we
 		// are probably dealing with mixed content). We handle also non-table content here.
+		// We just collapse selection to the first selected cell and pass non-table/mixed
+		// content to other paste handlers (#520).
```

Comment:
Actually this paste handler does not perform paste in such case. It just selects the first cell and ends its job – so I don't see my comment as outdated. Maybe it could be a little bit more detailed.